sst/SimengMemInterface.cc

```diff
@@ -0,0 +1,287 @@
+// clang-format off
+// DO NOT MOVE FROM TOP OF FILE - https://github.com/sstsimulator/sst-core/issues/865
+#include <sst/core/sst_config.h>
+// clang-format on
+
+#include "SimengMemInterface.hh"
+
+#include <iostream>
+
+using namespace SST::SSTSimeng;
+
+SimengMemInterface::SimengMemInterface(StandardMem* mem, uint64_t cl,
+                                       uint64_t max_addr, SST::Output* out)
+    : simeng::MemoryInterface() {
+  this->mem_ = mem;
+  this->clw_ = cl;
+  this->maxAddrMemory_ = max_addr;
+  this->output_ = out;
+};
+
+void SimengMemInterface::sendProcessImageToSST(const span<char> image) {
+  std::vector<uint8_t> data;
+  data.reserve(image.size());
+
+  for (size_t i = 0; i < image.size(); i++) {
+    data.push_back((uint8_t)image[i]);
+  }
+
+  StandardMem::Request* req = new StandardMem::Write(0, data.size(), data);
+  mem_->sendUntimedData(req);
+  return;
+};
+
+template <
+    typename T,
+    typename std::enable_if<std::is_base_of<
+        SimengMemInterface::SimengMemoryRequest, T>::value>::type* = nullptr>
```

Comment:
remove redefinition of default argument (already defined in header file) as it causes an error when compiled with Apple Clang i.e. `::type* = nullptr` -> `::type*`.